mist-lib/src/services/SignalTransductionService.js

```diff
@@ -0,0 +1,46 @@
+'use strict'
+
+const csv = require('csvtojson'),
+	path = require('path')
+
+const DefaultTableFilePath = path.resolve(__dirname, '..', 'signal-transduction', 'mist3-sig-trans-2017.csv')
+
+
+module.exports =
+class SignalTransductionService {
+	constructor(tableFilePath) {
+		this.tableFilePath = tableFilePath ? tableFilePath : DefaultTableFilePath
+		this.pfql = []
+		this.sigData = []
+	}
+
+	parseTable_() {
+		return new Promise((res, rej) => {
+			csv().fromFile(this.tableFilePath)
+				.on('json', (item) => {
+					this.pfql.push(this.makePFQLrule(item.domain_name, item.source))
+					this.sigData.push(item)
+				})
+				.on('end', () => {
+					res()
+				})
+				.on('error', (err) => {
+					rej(err)
+				})
+		})
+	}
+
```

Comment:
Add `setPfqlRules(pfqlRules)` / `pfqlRules()` methods for getting and setting the rules after instantiation.

Reply:
I don't understand this. If the constructor takes an optional array of PFQL rules, then there is no getting it. Right!?

I think what you mean is to initiate the PFQLStream. So this is what I did:
SignalTransductionService(rules) takes the PFQLRules to initiate.
and
`.setPfqlStream() to pass the rules and initialize the PFQLStream in this.pfqlStream to be used later.